### DISCUSSION TITLE
Problems with Aqara Opple switch getting bind with every other device on onboarding

### DISCUSSION THREAD

**Discussion by mcividino** [Koenkk/zigbee2mqtt#6136] [Devices]
I added 3 new aqara opple switches at home, but I'm having issues with them. As soon as they are onboarded, they control all the other zigbee devices in the house.
Even if I shut down the server, it keeps turning on and off everything.
If I press the onboarding button in the Aqara opple switches while the zigbee2mqtt is not running, it stops happening. However, as soon as I get zigbee2mqtt running, and onboard the switch again, the issue is back. 
I'm about to throw away these switches to buy three new ones, as I cannot use them.
Any idea of how to unbind the switches with the other devices? Also, how can I check the existing bindings?
Thanks in advance.

**Comment by chemaprados** [comment]
i have same problem

**Comment by hachel** [comment]
I have the exact same problem. That sounds exactly like the switch bound to all other devices, but under the devices bind tab I can only see a bind to the coordinator:
<img width="1913" alt="1" src="https://user-images.githubusercontent.com/7252437/133603529-01decad9-5099-4d31-86ec-b147b3fbbd44.png">
Unbinding it from any other device fails with "device_unbind_failed".

In the map it shoes the switch with a link to a bulb and no link to the router, but I don't know if that is a relevant to the problem:
<img width="1259" alt="2" src="https://user-images.githubusercontent.com/7252437/133604136-b0d6f0f5-ad9c-4bd3-8770-db2829b0ba9d.png">

EDIT: There are no groups under the groups tab.

**Comment by hachel** [comment]
Ok sorry for the hustle. As you can see above the bind to the coordinator is missing the LevelCtrl cluster. As soon as I selected it and hit bind it worked.

**Comment by tomillr** [comment]
I'm having the same experience trying to pair one of the newer Zigbee 3.0-based switches "Aqara wireless remote switch H1 (double rocker)" or "WRS-R02".

No matter how many time I've tried to reset and re-pair those switches, they automatically are bound to all the Zigbee lights in my house and not the hub itself.

Any idea how to properly pair them?

**Reply by d-c-u-p** [reply]
Can't help on that topic. 
But I hope you can answer a question to me. I'm looking for exactly this switch cos it suppossingly sends a "release" action after a long button press. Is that correct?

I'd need that to dim my lights but I can't find a suitable switch with release action :/

**Reply by d-c-u-p** [reply]
Just a thought on you actual question. Why would it matter to you if the switch connects to a "repeater" and not to the router itself?
Usually it connects to the point with the best connection. So to my understanding that should be just fine

**Comment by LorenzoBoers** [comment]
> I'm having the same experience trying to pair one of the newer Zigbee 3.0-based switches "Aqara wireless remote switch H1 (double rocker)" or "WRS-R02".
> 
> No matter how many time I've tried to reset and re-pair those switches, they automatically are bound to all the Zigbee lights in my house and not the hub itself.
> 
> Any idea how to properly pair them?

Having this same issue. You found a solution yet?

**Reply by hubjab** [reply]
> > I'm having the same experience trying to pair one of the newer Zigbee 3.0-based switches "Aqara wireless remote switch H1 (double rocker)" or "WRS-R02".
> > No matter how many time I've tried to reset and re-pair those switches, they automatically are bound to all the Zigbee lights in my house and not the hub itself.
> > Any idea how to properly pair them?
> 
> Having this same issue. You found a solution yet?

I had the same issue with Xiaomi/Aqara WRS-R02 and I solved it by using "Zigbee2mqtt Edge" version instead of "Zigbee2mqtt".

**Reply by mmakaay** [reply]
I got the same issue with RTS-R02. I haven't tried switching to the dev edge version. Does anybody have an idea about what the fix commit was?

**Reply by stefan242** [reply]
I have. See below!

**Comment by stefan242** [comment]
I have the same problem with several Aqara Opple switches. It works for a while with a new switch and then these switches turn on and off all zigbee bulbs in the mesh network. Even if I plug off the zigbee stick and turn off my Raspi with Node-RED. So it seems as if the switches connect directly to all bulbs it can find.

**Reply by stefan242** [reply]
I have solved my problem! If you use the

**operation_mode: event
in exposes of your device (wake up the device before changing modes!)**
you can use all of the double and triples button presses again and all of my switches do NOT switch other bulbs anymore.
Be sure, that event is marked.

**Reply by mmakaay** [reply]
Currently away on a little holiday, but back in a few days.
Something to look forward to, thanks for the update ;-)

**Comment by mikemoik** [comment]
Servus, i have the same problem. when debugging i got notice of those two messages published by the aquara opple switch (6 buttons, pushed was button 1, the friendly name is "but_op3_02"):
`object
topic: "zigbee2mqtt/but_op3_02"
payload: object
action: "button_1_single"
battery: 100
device: object
linkquality: 49
voltage: 3088
qos: 0
retain: false
_msgid: "4fbf7c07c9cdf7d8"`
which is the normal msg when single pushing button 1.

but there was also this msg:
`{"level":"debug","message":"Received Zigbee message from 'but_op3_02', type 'commandOff', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 0"}`

now, how to disable the behavior that the aquara opple switch is sending _commandOff_ to the _genOnOff_ cluster? it's doing the same thing when pushing button 2 only with sending _commandOn_. i think this is the root cause of this thread here.
honestly, i'm not able to understand what's going on here:
[https://www.zigbee2mqtt.io/guide/usage/binding.html#when-to-use-this](https://www.zigbee2mqtt.io/guide/usage/binding.html#when-to-use-this)

**Reply by stefan242** [reply]
What does your switch report in the tab state in Zigbee2MQTT?

```
{
    "battery": 100,
    "linkquality": 45,
    "operation_mode": "event",
    "voltage": 3174
}
```
operation_mode HAS to be "event"!
Other mode would be command and I assume, that is the setting of your switch!?

**Reply by mikemoik** [reply]
THAT did the trick, THANKS! maybe i overlooked it here in the thread!

**Comment by sebastianlorenzen** [comment]
The solution with the "event" config is still working with latest HA and Zigbee2MQTT version.

**Comment by IDarboIV** [comment]
Just had this myself last night, when repairing some Opple switches. Thought i was going mad, as it makes no sense. 

Will try this when im home and hopefully fixes the issue.
